10/calculator_test.go

```diff
@@ -0,0 +1,79 @@
+package calculator_test
+
+import (
+	"calculator"
+	"fmt"
+	"testing"
+)
+
+func TestSubtract(t *testing.T) {
+	testCases := []struct {
+		a, b int
+		want int
+	}{
+		{a: 1, b: 1, want: 0},
+		{a: 3, b: 2, want: 1},
+		{a: 6, b: 7, want: -1},
+	}
+	for _, testCase := range testCases {
+		got := calculator.Subtract(testCase.a, testCase.b)
+		if testCase.want != got {
+			t.Errorf("Add(%d, %d): want %d, got %d", testCase.a, testCase.b, testCase.want, got)
+		}
+	}
+}
+
+func TestDivide(t *testing.T) {
+	testCases := []struct {
+		a, b          int
+		want          int
+		errorExpected bool
+	}{
+		{a: 1, b: 1, want: 1, errorExpected: false},
+		{a: 4, b: 0, want: 2, errorExpected: true},
+		{a: 42, b: 7, want: 6, errorExpected: false},
+	}
+	for _, testCase := range testCases {
+		got, err := calculator.Divide(testCase.a, testCase.b)
+		if err == nil {
```

Comment:
There are actually four possible outcomes here, aren't there, each of which the test needs to handle:

1. We didn't expect an error, and we didn't get one, and `got == want`—succeed
2. We didn't expect an error, and we didn't get one, but `got != want`—fail
3. We didn't expect an error, but we did get one—fail
4. We _did_ expect an error, but we _didn't_ get one—fail

Try modifying either the code or the test to trigger each of these possible outcomes, and see if the test does the right thing in that case. (By 'the right thing', what I really mean is not only that the test should pass or fail as necessary, but that if it fails, it accurately explains what was wrong.)